tracing-opentelemetry/src/layer.rs

```diff
@@ -0,0 +1,276 @@
+where
+    S: Subscriber + for<'span> LookupSpan<'span>,
+    T: api::Tracer + 'static,
+{
+    /// Retrieve the parent OpenTelemetry [`SpanContext`] from the current
+    /// tracing [`span`] through the [`Registry`]. This [`SpanContext`]
+    /// links spans to their parent for proper hierarchical visualization.
+    fn parent_context(
+        &self,
+        attrs: &Attributes<'_>,
+        ctx: &Context<'_, S>,
+    ) -> Option<api::SpanContext> {
+        // If a span is specified, it _should_ exist in the underlying `Registry`.
+        if let Some(parent) = attrs.parent() {
+            let span = ctx.span(parent).expect("Span not found, this is a bug");
+            let mut extensions = span.extensions_mut();
+            extensions.get_mut::<api::SpanBuilder>().map(build_context)
+        // Else if the span is inferred from context, look up any available current span.
+        } else if attrs.is_contextual() {
+            ctx.current_span().id().and_then(|span_id| {
+                let span = ctx.span(span_id).expect("Span not found, this is a bug");
+                let mut extensions = span.extensions_mut();
+                extensions.get_mut::<api::SpanBuilder>().map(build_context)
+            })
+        // Explicit root spans should have no parent context.
+        } else {
+            None
+        }
+    }
+
+    /// Set the `OpenTelemetry` `Tracer` that this layer will use to produce
+    /// and track `Span`s.
+    ///
+    /// ```rust,no_run
+    /// use opentelemetry::{api::Provider, global, sdk};
+    /// use tracing_opentelemetry::OpenTelemetryLayer;
+    /// use tracing_subscriber::layer::SubscriberExt;
+    /// use tracing_subscriber::Registry;
+    ///
+    /// // Create a jaeger exporter for a `trace-demo` service.
+    /// let exporter = opentelemetry_jaeger::Exporter::builder()
+    ///     .with_agent_endpoint("127.0.0.1:6831".parse().unwrap())
+    ///     .with_process(opentelemetry_jaeger::Process {
+    ///         service_name: "trace_demo".to_string(),
+    ///         tags: Vec::new(),
+    ///     })
+    ///     .init().expect("Error initializing Jaeger exporter");
+    ///
+    /// // Build a provider from the jaeger exporter that always samples.
+    /// let provider = sdk::Provider::builder()
+    ///     .with_simple_exporter(exporter)
+    ///     .with_config(sdk::Config {
+    ///         default_sampler: Box::new(sdk::Sampler::Always),
+    ///         ..Default::default()
+    ///     })
+    ///     .build();
+    ///
+    /// // Get a tracer from the provider for a component
+    /// let tracer = provider.get_tracer("component-name");
+    ///
+    /// // Create a layer with the configured tracer
+    /// let otel_layer = OpenTelemetryLayer::with_tracer(tracer);
+    ///
+    /// // Use the tracing subscriber `Registry`, or any other subscriber
+    /// // that impls `LookupSpan`
+    /// let _subscriber = Registry::default()
+    ///     .with(otel_layer);
```

Comment:
```suggestion
    /// let subscriber = Registry::default()
    ///     .with(otel_layer);
    /// # drop(subscriber);
```
or something, so it isn't "unused" in the example?